addons/test_mass_mailing/tests/test_performance.py

```diff
@@ -91,8 +101,10 @@ def test_send_mailing_w_bl(self):
         })
 
         # runbot needs +62 compared to local
-        with self.assertQueryCount(__system__=1942, marketing=1942):
+        with self.assertQueryCount(__system__=1946, marketing=1948), \
+             patch.object(IrMailServer, '_find_mail_server', side_effect=find_mail_server) as patched_find_mail_server:
             mailing.action_send_mail()
 
+        self.assertEqual(patched_find_mail_server.call_count, 1, 'Must be called only once')
```

Comment:
This does not really belong to a performance test.

Reply:
In a sense yes, because it's important to check that we call `find_mail_server` only once when we send email in batch...(because someone could easily "simplify" the code by calling it once per email, and we should ensure that it never happen)

Reply:
Yes but performance tests are generally there for query count. I would move that test in another file to keep performance file for queries only.

Reply:
moved to `test_mass_mailing_server_choice` in `test_mail_full::test_mass_mailing`